### ISSUE TITLE
Add information on versions of MNI template in fetchers and user guide

### ISSUE THREAD

**Issue by NicolasGensollen** [nilearn/nilearn#3075]
See discussion #3074. 

We need to better explain:

- the fact that the atlases were generated using a specific version of the MNI template (for example by listing that version in the docstrings of the fetchers)
- explain to the user that they need to ensure that their data are aligned to the same version (we can add a note in the docstrings but also in the user guide, for example [here](https://nilearn.github.io/stable/manipulating_images/manipulating_images.html) and [here](https://nilearn.github.io/stable/connectivity/functional_connectomes.html) as pointed out by @emdupre in #3074.

**Comment by tsalo** [comment]
Per today's call, a few other elements to include in the corresponding PR:

- Add a pointer to templateflow in the documentation.
- Explain that nilearn's scope is not to provide a general data fetcher, they should only be used for examples.
- Remove from docs that we welcome new atlases.

**Comment by NicolasGensollen** [comment]
Thanks @tsalo ! :+1:

**Comment by psadil** [comment]
FWIW, I'd find this kind of information very helpful. For example, `fetch_coords_power` mentions "MNI" with a link to the glossary: https://github.com/nilearn/nilearn/blob/faad5c7df045cf616bec55795db6dcd9a85f140a/doc/glossary.rst?plain=1#L187-L191

I take that to mean the coordinates are in the "MNI152Lin" space, using this terminology: https://bids-specification.readthedocs.io/en/latest/appendices/coordinate-systems.html. But I'm not sure, because, for example, the Schaefer atlas is described as being in MNI152: https://github.com/nilearn/nilearn/blob/faad5c7df045cf616bec55795db6dcd9a85f140a/nilearn/datasets/atlas.py#L2068-L2069

I think the Schaefer atlas comes from MNI152NLin6ASym, but I don't know and it seems easy to make a mistake.

**Comment by Remi-Gau** [comment]
Yeah I think we can try to track what space was used for each atlas but given the confusion in the field around "space / template / atlas" I would not be surprised if for quite a few of them we get an answer like "some sort of MNI space".

**Comment by psadil** [comment]
Is there a preference to have a pull request be all or nothing vs something incremental? At least a few may be straightforward.

**Comment by Remi-Gau** [comment]
> be all or nothing vs something incremental

if you are adding 2-3 lines that are more or less the same in 30 different files, it can be one PR

for now I'd prefer to have the dataset rst files updated but on the long term I would not mind having the space be a key in the dictionary / bunch returned by the fetcher

**Comment by Remi-Gau** [comment]
@psadil I am working on this in #5174 
if you want to cross check the info I am adding about the template used for each atlas

**Comment by psadil** [comment]
Oh wow, [these look](https://github.com/nilearn/nilearn/pull/5174#issuecomment-2678809506) great!

**Comment by Remi-Gau** [comment]
> * the fact that the atlases were generated using a specific version of the MNI template (for example by listing that version in the docstrings of the fetchers)

most atlas fetchers (hopefully all) should return a sklearn Bunch where one of the key is "template" that describres which template the atlas uses.

**Comment by Remi-Gau** [comment]
just to keep things updated:
the following atlas have unspecified template:

- fetch_atlas_pauli_2017
- fetch_atlas_craddock_2012
- fetch_atlas_msdl
- fetch_atlas_fsl_Juelich
- fetch_atlas_smith_2009

**Comment by bthirion** [comment]
msdl was likely based on SPM-preprocessed data so it should be the default SPM template.

**Comment by emdupre** [comment]
@Remi-Gau can this be closed ?
